### PR TITLE
Fake sceNpMatching2GetLobbyInfoList

### PR DESCRIPTION
**Pull request by RipleyTom**
Fixes https://github.com/RPCS3/rpcs3/issues/15890

Very few games use that API:
```
Dark Souls [BLES01396, BLUS30782]
Guilty Gear Xrd -Revelator- [NPUB31806]
Guilty Gear Xrd -SIGN- [NPUB31555]
Guilty Gear Xrd: Sign [BLUS31467]
Gundam Viewchives [NPJB90232]
Metal Slug 2 [NPEB00681]
頭文字D EXTREME STAGE [BLJM60055]
```

InitialD actually only uses that one lobby function and works with the fake values returned in the structure.
I haven't tested the other games(I think the Guilty Gear games require an actual implementation).

### CODE REVIEW

File: rpcs3/Emu/NP/np_requests.cpp [hunk 1]
```diff
@@ -718,6 +718,39 @@ namespace np
...
+
+		u32 event_key = get_event_key();
+
+		auto& edata = allocate_req_result(event_key, SCE_NP_MATCHING2_EVENT_DATA_MAX_SIZE_GetLobbyInfoList, sizeof(SceNpMatching2GetLobbyInfoListResponse));
+		auto* resp = reinterpret_cast<SceNpMatching2GetLobbyInfoListResponse*>(edata.data());
+
+		resp->range.size = 1;
+		resp->range.startIndex = 1;
+		resp->range.total = 1;
+		auto lobby_data = edata.allocate<SceNpMatching2LobbyDataExternal>(sizeof(SceNpMatching2LobbyDataExternal), resp->lobbyDataExternal);
```
Comment (Megamouse) [inline rpcs3/Emu/NP/np_requests.cpp:740]: Unused?

**Comment by RipleyTom** [comment]
I added a debug print for the request and while I was at it renamed all those print functions to be more standardized.

File: rpcs3/Emu/NP/np_structs_extra.cpp [hunk 2]
```diff
@@ -387,11 +387,23 @@ namespace extra_nps
...
 		}
 	}
 
+	void print_SceNpMatching2GetLobbyInfoListRequest(const SceNpMatching2GetLobbyInfoListRequest* resp)
+	{
+		sceNp2.warning("SceNpMatching2GetLobbyInfoListRequest:");
+		sceNp2.warning("worldId: %d", resp->worldId);
+		print_SceNpMatching2RangeFilter(&resp->rangeFilter);
+
+		for (u32 i = 0; i < resp->attrIdNum; i++)
```
Comment (Megamouse) [inline rpcs3/Emu/NP/np_structs_extra.cpp:401]:
```suggestion
		for (u32 i = 0; i < resp->attrIdNum && resp->attrId; i++)
```

**Comment by CrazyCSIW6** [comment]
Since this change was merged, GTA V has been freezing constantly when going online on 1.06-1.12

**Pull request merged**